akka-stream-tests/src/test/scala/akka/stream/scaladsl/GraphMergeSortedSpec.scala

```diff
@@ -40,6 +43,38 @@ class GraphMergeSortedSpec extends TwoStreamsSetup with GeneratorDrivenPropertyC
       }
     }
 
+    "work as a special case of MergeSortedN with N=2" in {
+      val gen = Gen.listOf(Gen.oneOf(false, true))
+
+      forAll(gen) { picks ⇒
+        val N = picks.size
+        val (left, right) = picks.zipWithIndex.partition(_._1)
+        Source.mergeSortedN(List(Source(left.map(_._2)), Source(right.map(_._2))))
+          .grouped(N max 1)
+          .concat(Source.single(Nil))
+          .runWith(Sink.head)
+          .futureValue should ===(0 until N)
+      }
+    }
+
+    "work" in {
+      // List(false, false, false, true, false, true, false, true, true, false)
+      val leftSource: Source[Int, NotUsed] = Source(List(0, 2))
+      val rightSource: Source[Int, NotUsed] = Source(List(1))
+
+      val underTest: Source[Int, NotUsed] = Source.mergeSortedN(List(leftSource, rightSource))
+
+      val grouped: Source[immutable.Seq[Int], NotUsed] = underTest
+        .grouped(4)
+
+      val sortedResult: immutable.Seq[Int] = grouped
+        .concat(Source.single(Nil))
+        .runWith(Sink.head)
+        .futureValue
+
+      sortedResult shouldBe List(0, 1, 2)
+    }
+
```

Comment:
Some things I think it could be good to cover explicitly:
* when one upstream fails / is failed to start with
* when you feed it one source that never emits + one that does emit
* when you feed it one source that is already/immediately completed

The stream testkit probes are useful to write such tests.

Reply:
I think these cases are now all covered:
```
"GraphMergeSortedN" must {
"work with one immediately failed and one nonempty publisher"
"work with one delayed failed and one nonempty publisher"
"never emit if a source does not emit"
"work with one immediately completed and one nonempty publisher"
}
```